File: sdk/src/main/java/com/cxense/cxensesdk/model/WidgetVisibilityReport.java
package com.cxense.cxensesdk.model;

import androidx.annotation.NonNull;

import com.cxense.cxensesdk.Preconditions;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * Request object for reporting widgets visibilities to server.
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2019-05-14).
 */
@SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
public class WidgetVisibilityReport {
    @SerializedName("impressions")
    List<Impression> impressions;

    public WidgetVisibilityReport(@NonNull Collection<Impression> impressions) {
        Preconditions.checkForNull(impressions, "impressions");
        this.impressions = new ArrayList<>(impressions);
    }
}

File: sdk/src/main/java/io/piano/android/cxense/model/WidgetVisibilityReport.kt
package io.piano.android.cxense.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

/**
 * Request object for reporting widgets visibilities to server.
 *
 */
@JsonClass(generateAdapter = true)
class WidgetVisibilityReport(
    @Json(name = "impressions") val impressions: List<Impression>,
)
